bundles/org.openhab.binding.sony/src/main/java/org/openhab/binding/sony/internal/scalarweb/ScalarWebClientFactory.java

```diff
@@ -0,0 +1,213 @@
+     * @return a {@link ScalarWebClient}
+     * @throws IOException if an IO Exception occurs
+     * @throws ParserConfigurationException if a parser configuration exception occurrs
+     * @throws SAXException if a SAX exception occurs readonly the documents
+     * @throws URISyntaxException if a URI syntax exception occurs
+     */
+    public ScalarWebClient get(final URL scalarWebUrl, final ScalarWebContext context)
+            throws IOException, ParserConfigurationException, SAXException, URISyntaxException {
+        Objects.requireNonNull(scalarWebUrl, "scalarWebUrl cannot be null");
+        Objects.requireNonNull(context, "context cannot be null");
+
+        if (StringUtils.isEmpty(scalarWebUrl.getPath())) {
+            return getDefaultClient(scalarWebUrl, context);
+        } else {
+            return queryScalarWebSclient(scalarWebUrl, context);
+        }
+    }
+
+    /**
+     * Helper method to attempt to get a 'default' client. Basically if we have just an ip address, try to get a client
+     * base on querying it
+     * 
+     * @param scalarWebUrl a non-null URL
+     * @param context a non-null context
+     * @return a {@link ScalarWebClient}
+     * @throws IOException if an IO Exception occurs
+     * @throws DOMException if a DOM exception occurs readonly the documents
+     * @throws URISyntaxException if a URI syntax exception occurs
+     */
+    private ScalarWebClient getDefaultClient(final URL scalarWebUrl, final ScalarWebContext context)
+            throws DOMException, IOException, URISyntaxException {
+        Objects.requireNonNull(scalarWebUrl, "scalarWebUrl cannot be null");
+        Objects.requireNonNull(context, "context cannot be null");
+
+        final URL homeAudioUrl = new URL(scalarWebUrl.getProtocol(), scalarWebUrl.getHost(), HOME_AUDIO_PORT,
+                LIKELY_GUIDE_PATH);
+        logger.debug("Testing Default Scalar Web client to see if it's a home audio device (AVR/Soundbar): {}",
+                homeAudioUrl);
+        try (SonyTransport transport = SonyTransportFactory.createHttpTransport(homeAudioUrl,
+                GsonUtilities.getApiGson())) {
+            final ScalarWebResult homeAudioRes = transport
+                    .execute(new ScalarWebRequest(1, ScalarWebMethod.GETVERSIONS, ScalarWebMethod.V1_0));
+            if (homeAudioRes.getHttpResponse().getHttpCode() == HttpStatus.OK_200) {
+                final URL baseUrl = new URL(scalarWebUrl.getProtocol(), scalarWebUrl.getHost(), HOME_AUDIO_PORT,
+                        LIKELY_PATH);
+                return new ScalarWebClient(new ScalarWebDeviceManager(baseUrl, context));
+            }
+        }
+
+        final URL personalAudioUrl = new URL(scalarWebUrl.getProtocol(), scalarWebUrl.getHost(), PERSONAL_AUDIO_PORT,
+                LIKELY_GUIDE_PATH);
+        logger.debug("Testing Default Scalar Web client to see if it's a personal audio device (Wireless speaker): {}",
+                homeAudioUrl);
```

Comment:
```suggestion
                personalAudioUrl);
```
These two blocks of code are virtually identical, I would suggest making a function and just parameterize the differences.